### ISSUE TITLE
Render paths similar like the Carto-Style (distinguish between asphalt/concrete and other surfaces)

### ISSUE THREAD

**Issue by gerion0** [organicmaps/organicmaps#8485]
**Describe the issue**
When using OM for inline skating or speed biking, it is crucial to be able to distinguish between asphalt/concrete roads and roads without a hard surface. I saw that the release notes of the may release stated that OM has changed its style here:
> more consistency in displaying paved paths as white dashed lines, unpaved ones as brown dashed

However, for paths, this seems not to work. Currently (as a workaround), I open osm.org in a browser in parallel, since the CartoCSS style there displays these paths differently.

**Steps to reproduce**
Take this path: https://www.openstreetmap.org/way/24991968

This is the rendering on the OSM website:
![image](https://github.com/organicmaps/organicmaps/assets/32870336/b441193c-0b1d-49db-919d-180afa8ea821)
This is the rendering in OM 2024.06.02-12-FDroid:
![Screenshot_20240617-214431_Organic Maps](https://github.com/organicmaps/organicmaps/assets/32870336/76963a46-baab-43d6-b6ad-2f48be0278d0)

**Expected behaviour**
I would expect, that there is a visual difference between https://www.openstreetmap.org/way/24991968 and https://www.openstreetmap.org/way/213820851 in OM.

**System information:**
 - Operating system and its version: Android 14
 - Organic Maps version: 2024.06.02-12-FDroid

**Additional context**
These issues might be relevant. However, I found none of them really matching: #7139 #6285 #8391 #5967 #6694 #8231

**Comment by matkoniecz** [comment]
note: OSM Carto is using tractype but moving to `surface` was being considered.

Using `surface` should work well here, maybe with `tracktype` used as fallback.

**Comment by matkoniecz** [comment]
> However, for paths, this seems not to work

AFAIK it works for paths (`highway=path` `footway`) but it does not work for tracks (`highway=track`)

**Comment by gerion0** [comment]
It may be that Carto uses also the `grade` specification. https://www.openstreetmap.org/way/461362188 for example has only a grade specific and is marked the same way like asphalt streets.

And you are right, I mixed up paths and tracks. `highway=track` is the correct classification here.

**Comment by matkoniecz** [comment]
> It may be that Carto uses also the grade specification. https://www.openstreetmap.org/way/461362188 for example has only a grade specific and is marked the same way like asphalt streets.

yes, as mentioned in https://github.com/organicmaps/organicmaps/issues/8485#issuecomment-2176158543

**Comment by matkoniecz** [comment]
> And you are right, I mixed up paths and tracks. highway=track is the correct classification here.

Can you change issue title?

**Comment by pastk** [comment]
Related to
- https://github.com/organicmaps/organicmaps/pull/7878

Doing it for this specific case of paved tracks is not hard. I guess we'll render them as thin white lines similar to how highway=service are rendered. WDYT?

**Comment by pastk** [comment]
Doing an opposite conversion - from unpaved roads to "tracks" (i.e. something that looks its unpaved) is more tricky. We can't just render all of them as dark dashes (like tracks are rendered now), because we'll lose information about importance and intended use of the road.

E.g. consider a village where all roads are unpaved, but some of them are still "main" (and e.g. connecting to other settlements).
So they should be rendered similar to other same class roads (e.g. tertiary), but somehow distinct to hint about road surface/quality.
E.g. for the tertiary it could be a white line with a thinner dark dashed pattern over it?
For the secondary - a yellow-orange line with a thinner dark dashed, etc.

WDYT? Ideas are welcome!

**Comment by Misalf-git** [comment]
Tracks could maybe be rendered as solid (non-dashed) lines to differentiate them from paths?

Areas familiar to me, where `=track`s and `=path`s are mapped (in the woods, not belonging to the actual road network), their distinction might make sense.
Those `=track`s are not meant for public motorized traffic though, so having them rendered like `highway=service` wouldn't seem right to me.

**Comment by pastk** [comment]
> Areas familiar to me, where `=track`s and `=path`s are mapped (in the woods, not belonging to the actual road network), their distinction might make sense.
> Those `=track`s are not meant for public motorized traffic though, so having them rendered like `highway=service` wouldn't seem right to me.

We're talking about paved tracks here, right?

And many service roads (alleys, driveways to private properties and industrial objects, etc.) are also not meant / intended for wide public use, right?

**Comment by pastk** [comment]
> Tracks could maybe be rendered as solid (non-dashed) lines to differentiate them from paths?

It feels like a big visual change, but it might work, its worth exploring it more IMO.

In this case a solid line will signify that the way is meant for wheeled vehicles, while something dashed doesn't fit.

**Comment by pastk** [comment]
Actually there 2 related questions here:
1. How to distinguish between misc "importance" classes of the roads (secondary, residential, service, track..). Its something that could be not easy to tell by a user by just being on the road.
2. How to distinguish between quality / surface of the roads (paved, unpaved, very bad smoothness or track grade). Something very easy to tell once on the road. And users expect to be able to read it off the map. (Now its like solid white or orange line is paved, while dark dashed is unpaved - but doesn't work for all cases.)

**Comment by pastk** [comment]
Qaulity-wise it'd be nice to have ~3 grades similar to 
- https://github.com/organicmaps/organicmaps/pull/7926
E.g. "a good road" (a good track could fit it also), "ok, but low speed" (a bad primary could fit it also), "4x4, better know what you're doing".

I was thinking to use shorter/longer dashed patterns to differentiate (i.e. short dashes with longer gaps signify bad quality). In case of non-track roads the pattern is meant to be overlayed over solid line.

But if we make tracks render as solid lines, then how do we differentiate quality?
By color?

**Comment by Misalf-git** [comment]
> > Areas familiar to me, where `=track`s and `=path`s are mapped (in the woods, not belonging to the actual road network), their distinction might make sense.
> > Those `=track`s are not meant for public motorized traffic though, so having them rendered like `highway=service` wouldn't seem right to me.
> 
> We're talking about paved tracks here, right?

Sorry, paved tracks are rare there, I meant unpaved / compacted tracks. White for paved does make sense.

> In this case a solid line will signify that the way is meant for wheeled vehicles, while something dashed doesn't fit.

Exactly. Currently, a small muddy path that is as narrow as a footstep and a 5 meters wide, compacted grade-2 track look the same.

**Comment by pastk** [comment]
> Exactly. Currently, a small muddy path that is as narrow as a footstep and a 5 meters wide, compacted grade-2 track look the same.

You mean the distinction between a `highway=path` and a `highway=track`, don't you?

They're rendered distinct enough, no?
![image](https://github.com/user-attachments/assets/98d34135-efbc-4066-bb84-059a2970ed59)

**Comment by Misalf-git** [comment]
You're right... I confused something.
Well, sorry for that!

**Comment by dealerofsalvation** [comment]
> We got 3.33 EUR donation with a comment:
> 
> > Please implement Github issue 8485. It's essential for cycling and skating. Thanks

That was me. I support this project from now on anyway, whether you implement the issue or not, with an automatic monthly payment of 3.33 EUR. If the issue gets implemented, I'll change the comment :)

**Comment by useross** [comment]
This would be a useful implementation, especially for cycling.
